cluster/cluster.go

```diff
@@ -563,3 +571,101 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
 	gplog.FatalOnError(err)
 	return segConfigs
 }
+
+// GetSegmentConfigurationFromFile function to read segment configuration from dumpconfiguration file.
+// file should have the absolute path else if the file is empty it will generate default path coordinator_data_dir/gpsegconfig_dump
+// file should have the valid configuration format
+// dbid content role prole mode status port hostname address datadir
+func GetSegmentConfigurationFromFile(file string) ([]SegConfig, error) {
+	var conFilePath string
+	if file == "" {
+		filename := "gpsegconfig_dump" // filename which dump from fts process to master datadir
+		coordinator_datadir := os.Getenv("COORDINATOR_DATA_DIRECTORY")
+		if coordinator_datadir == "" {
+			coordinator_datadir = os.Getenv("MASTER_DATA_DIRECTORY")
+		}
+		if coordinator_datadir == "" {
+			errStr := "COORDINATOR_DATA_DIRECTORY is not set, not able to generate gpsegconfig_dump file path"
+			gplog.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+		}
+		conFilePath = filepath.Join(coordinator_datadir, filename)
+	} else {
+		conFilePath = file
+	}
+
+	fd, err := os.Open(conFilePath)
+	if err != nil {
+		errStr := fmt.Sprintf("failed opening file %s : %s", conFilePath, err.Error())
+		gplog.Error(errStr)
+		return nil, fmt.Errorf(errStr)
+	}
+	defer fd.Close()
+
+	results := make([]SegConfig, 0)
+	scanner := bufio.NewScanner(fd)
+
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		parts := len(fields)
+		var datadir string
+
+		/* older version of gpsegconfig_dump file has 9 parts as it doesn't have datadir
+		1 -1 p p n u 7000 shrakeshSMD6M.vmware.com shrakeshSMD6M.vmware.com */
+		/* newer version of gpsegconfig_dump file has 10 parts as it does have datadir
+		1 -1 p p n u 7000 shrakeshSMD6M.vmware.com shrakeshSMD6M.vmware.com /data/qddir/demoDataDir-1 */
+		if parts != 9 && parts != 10 {
+			errStr := fmt.Sprintf("Unexpected number of fields (%d) in line: %s", parts, scanner.Text())
+			gplog.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+		}
+
+		dbID, err := strconv.Atoi(fields[0])
+		if err != nil {
+			errStr := fmt.Sprintf("error converting dbID %s", err.Error())
+			gplog.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+		}
+
+		content, err := strconv.Atoi(fields[1])
+		if err != nil {
+			errStr := fmt.Sprintf("error converting content %s", err.Error())
+			gplog.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+		}
+
+		port, err := strconv.Atoi(fields[6])
+		if err != nil {
+			errStr := fmt.Sprintf("error converting port %s", err.Error())
+			gplog.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+		}
+
+		if parts == 10 { //for new version of gpsegconfig_dump file
+			datadir = fields[9]
+		}
+
+		seg := SegConfig{
+			DbID:      dbID,
+			ContentID: content,
+			Role:      fields[2],
+			PRole:     fields[3],
+			Status:    fields[4],
+			Mode:      fields[5],
+			Address:   fields[8],
+			Port:      port,
+			Hostname:  fields[7],
```

Comment:
We can move this to above the "Address" field to maintain the sequence

Reply:
updated the sequence of all the fields to make it consistent with the gp_segment_configuration table output.